client/src/app-header/AppHeaderUser.js

```diff
@@ -11,9 +11,15 @@ function mapStateToProps(state) {
 const Connected = connect(mapStateToProps)(React.memo(UserButton));
 
 function UserButton({ currentUser }) {
-  return (
-    <div className={styles.style}>{currentUser.name || currentUser.email}</div>
-  );
+  if(currentUser._id === "noauth") {
+  	return null
+  }
+  else {
```

Comment:
This a readability preference in general, but since the `if` block above returns and this `else` block is not conditioned, the `else` can be removed and simply return the `<div ...`.

So like this:

```jsx
if (currentUser._id === "noauth") {
  return null;
}

return <div className={styles.style}> {currentUser.name || currentUser.email}</div>;
```

Reply:
✅